### PR TITLE
Updated sqlite3 to version 3.10.2

### PR DESCRIPTION
**Pull request by biozic**

### CODE REVIEW

File: etc/c/sqlite3.d [hunk 1]
```diff
@@ -1604,14 +1660,17 @@ enum
     SQLITE_TESTCTRL_VDBE_COVERAGE           = 21,
     SQLITE_TESTCTRL_BYTEORDER               = 22,
     SQLITE_TESTCTRL_ISINIT                  = 23,
-    SQLITE_TESTCTRL_LAST                    = 23,
+    SQLITE_TESTCTRL_SORTER_MMAP             = 24,
+    SQLITE_TESTCTRL_IMPOSTER                = 25,
+	SQLITE_TESTCTRL_LAST                    = 25,
 }
```
Comment (schuetzm) [inline etc/c/sqlite3.d]: nitpick: identation is off here

File: etc/c/sqlite3.d [hunk 2]
```diff
@@ -1383,12 +1436,16 @@ struct sqlite3_index_info
...
 enum
 {
+    SQLITE_INDEX_SCAN_UNIQUE      = 1,
     SQLITE_INDEX_CONSTRAINT_EQ    = 2,
     SQLITE_INDEX_CONSTRAINT_GT    = 4,
     SQLITE_INDEX_CONSTRAINT_LE    = 8,
     SQLITE_INDEX_CONSTRAINT_LT    = 16,
     SQLITE_INDEX_CONSTRAINT_GE    = 32,
-    SQLITE_INDEX_CONSTRAINT_MATCH = 64
+    SQLITE_INDEX_CONSTRAINT_MATCH = 64,
```
Comment (schuetzm) [inline etc/c/sqlite3.d]: please realign the `=` above

File: etc/c/sqlite3.d [hunk 3]
```diff
@@ -722,7 +735,8 @@ enum
     SQLITE_LIMIT_ATTACHED                  = 7,
     SQLITE_LIMIT_LIKE_PATTERN_LENGTH       = 8,
     SQLITE_LIMIT_VARIABLE_NUMBER           = 9,
-    SQLITE_LIMIT_TRIGGER_DEPTH            = 10
+    SQLITE_LIMIT_TRIGGER_DEPTH            = 10,
+	SQLITE_LIMIT_WORKER_THREADS           = 11,
 }
```
Comment (schuetzm) [inline etc/c/sqlite3.d]: wrong indentation

File: etc/c/sqlite3.d [hunk 4]
```diff
@@ -1854,6 +1914,51 @@ enum
     SQLITE_REPLACE  = 5
 }
 
+/*
+** CAPI3REF: Prepared Statement Scan Status Opcodes
+*/
+enum
```
Comment (MartinNowak) [inline etc/c/sqlite3.d]: From the tester 'etc/c/sqlite3.d' contains trailing whitespace at line 1920.

**Comment by MartinNowak** [comment]
RLGTM

**Comment by biozic** [comment]
I kept the "pointer to constant" C syntax that was already there, for instance `const X *` parameters. But the syntax should be `const(X)*` on the D side. I don't know what the policy is. Should I change this?

**Comment by MartinNowak** [comment]
Auto-merge toggled on

**Pull request merged**